diagnostics-extension/src/sw.ts

```diff
@@ -6,8 +6,19 @@
  * @fileoverview Service worker script
  */
 
+import { Request, RequestScope } from '@common/message';
+import { handleTelemetry } from './telemetry.controller';
+
 chrome.runtime.onInstalled.addListener(
   (details: chrome.runtime.InstalledDetails) => {
     console.log('Service worker is installed!', details);
   }
 );
+
+chrome.runtime.onMessageExternal.addListener((req: Request, sender, res) => {
+  console.log(req, sender);
+  switch (req.scope) {
+    case RequestScope.TELEMETRY:
```

Comment:
also add other request scopes here and throw an error with something like "not implemented" message.